packages/nextjs/utils/scaffold-eth/networks.ts

```diff
@@ -160,21 +159,19 @@ export function getBlockExplorerTxLink(network: Network, txnHash: string) {
   return blockExplorerTxURL;
 }
 
-export const getNetworkDetailsByChainId = (chainId: number) =>
-  Object.values(NETWORKS).find(val => val.chainId === chainId);
+export const getNetworkDetailsByChainId = (id: number) => Object.values(NETWORKS).find(val => val.id === id);
 
 /**
  * @dev Returns the target network metadata or defaults to hardhat if the network is unsupported/undefined
  */
 export const getTargetNetwork = () => {
   const network = process.env.NEXT_PUBLIC_NETWORK;
-  const { chains } = appChains;
-  const configuredChain = chains.find(chain => chain.network === network);
+  const configuredChain = NETWORKS[network ?? "hardhat"];
```

Comment:
Note: It will be Hardhat when there is no `NEXT_PUBLIC_NETWORK` in env, but `undefined` when the network is `''`, (`NEXT_PUBLIC_NETWORK=` in `.env`)

Reply:
++, but it will be set again to `hardhat` if it's `undefined`....but maybe we refactor to make it simple

Reply:
Yep, and it will not show console error in the first case, but it will show it in second

Reply:
yes ! 🙌, Fix at #187